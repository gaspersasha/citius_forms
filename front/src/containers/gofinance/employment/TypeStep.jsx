import React from 'react';
import PropTypes from 'prop-types';

import { TapButtons } from '~components';
import { FORM } from '~constants';
import s from './employment.module.sass';

const { EMPLOYMENT_TYPES } = FORM;

const TypeStep = ({ onChange, employmentType }) => (
  <div className={s.typeFormWrapper}>
    <TapButtons
      label="What is your employment type?"
      subTitle="Select from the list below"
      options={EMPLOYMENT_TYPES}
      onChange={onChange}
      required
      status={employmentType.status}
      value={employmentType.value}
    />
  </div>
);

TypeStep.propTypes = {
  onChange: PropTypes.func.isRequired,
  employmentType: PropTypes.shape({
    status: PropTypes.string,
    value: PropTypes.string,
  }).isRequired,
};

export default TypeStep;
