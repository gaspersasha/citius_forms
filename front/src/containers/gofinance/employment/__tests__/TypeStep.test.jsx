import React from 'react';
import { INPUT_STATUS } from '~constants';
import TypeStep from '../TypeStep';

describe('<TypeStep />', () => {
  const props = {
    onChange: jest.fn(),
    employmentType: {
      value: 'FULL_TIME_PERMANENT',
      status: INPUT_STATUS.VALID,
    },
  };

  const wrapper = shallow(<TypeStep {...props} />);

  it('renders without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
