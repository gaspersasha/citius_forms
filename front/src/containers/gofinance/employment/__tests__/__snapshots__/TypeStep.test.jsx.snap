// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeStep /> renders without errors 1`] = `
<div
  className="typeFormWrapper"
>
  <TapButtons
    label="What is your employment type?"
    name=""
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "option": "Full time permanent",
          "value": "FULL_TIME_PERMANENT",
        },
        Object {
          "option": "Part time permanent",
          "value": "PART_TIME_PERMANENT",
        },
        Object {
          "option": "Armed services",
          "value": "ARMED_SERVICES",
        },
        Object {
          "option": "Self employed",
          "value": "SELF_EMPLOYED",
        },
        Object {
          "option": "Retired",
          "value": "RETIRED",
        },
        Object {
          "option": "Education",
          "value": "EDUCATION",
        },
        Object {
          "option": "Homemaker",
          "value": "HOMEMAKER",
        },
        Object {
          "option": "Carer",
          "value": "CARER",
        },
        Object {
          "option": "Benefits",
          "value": "BENEFITS",
        },
        Object {
          "option": "Unemployed",
          "value": "UNEMPLOYED",
        },
        Object {
          "option": "Temporary / Contract",
          "value": "TEMPORARY_CONTRACT",
        },
      ]
    }
    required={true}
    status="valid"
    subTitle="Select from the list below"
    value="FULL_TIME_PERMANENT"
  />
</div>
`;
